pyrokinetics/metadata.py

```diff
@@ -0,0 +1,44 @@
+import uuid
+from datetime import datetime
+from typing import Dict
+
+try:
+    from importlib.metadata import version, PackageNotFoundError
+except ModuleNotFoundError:
+    from importlib_metadata import version, PackageNotFoundError
+try:
+    __version__ = version(__name__)
+except PackageNotFoundError:
+    try:
+        from setuptools_scm import get_version
+
+        __version__ = get_version(root="..", relative_to=__file__)
+    except ImportError:
+        __version__ = "0.0.1"
+
+
+# Define UUID and session start as module-level variables.
+# Determined at the first import, and should be fixed during each session.
+
+__uuid = uuid.uuid4()
+__session_start = datetime.now()
```

Comment:
I think these probably want to be instance-level variables on `Pyro`, rather than globals. Actually, probably the time wants to be set when we write files too

Reply:
My intent with those globals was to use them to identify which Python sessions created each object. I'm imagining a scenario where a bunch of files are generated in a parameter scan of some sort, which may involve multiple `Pyro` objects (or no Pyro objects at all), and these UUIDs/session starts can be used to tie them all together. I think a `pyro_uuid` would also be helpful, but `Pyro` doesn't use `metadata` yet, and I think that'd be best achieved in another PR. I'm currently not sure how a `Pyro` could pass this down to every object it handles, and it might require changing constructors/function signatures throughout the code.

Reply:
Actually, I may need to rethink how this works altogether. Currently the metadata is set when an `Equilibrium` or `Numerics` is created, and the same would be pushed to `GKOutput`/`Kinetics`/`LocalGeometry` if they adopt `DatasetWrapper` as a base class. If it'd be better to only add this when we write to files, and if we'd need it to be handled (exclusively?) by `Pyro`, then the current system just won't work. Could this be handled later in a different Issue/PR?

Reply:
Yes, we can put it off for now. Something is better than nothing, but it's probably going to require a fair bit of thought to get this right